lib/keyfile/keyfile.c

```diff
@@ -333,6 +326,10 @@ read_base64(const char * keybuf, size_t keylen, uint64_t * machinenum,
 		keybuf += llen;
 		keylen -= llen;
 
+		/* If there was a missing EOL, don't skip past the EOL. */
```

Comment:
I would describe the situation as EOF rather than "missing EOL," but it is not very different.